app/assets/javascripts/components/signup.js

```diff
@@ -0,0 +1,49 @@
+$( document ).ready(function() {
+
+  $('.error').hide();
+
+  $(".signupForm").submit(function(event) {
+
+  var errorMessage = "";
+
+  event.preventDefault();
+
+  function isValidFullName(fullName) {
+    var validate_name = new RegExp (/^[a-zA-Z ]+$/);
+    return validate_name.test(fullName);
+  }
```

Comment:
The `isValid...` functions can be scoped outside the signup form submit function.  You'll probably want to reuse some these on the login form.